NOMIS/nomis.go

```diff
@@ -301,3 +307,25 @@ func downloadFile() {
 
 	fmt.Printf("Downloaded a file %s with size %d", fileName, size)
 }
+
+func checkSubString (res CenStructure, index0, indx int)  string {
+	var strArray []string
+	var result string
+	value :=strings.Contains(res.Structure.Keyfamilies.Keyfamily[index0].Annotations.Annotation[indx].Text.(string), "http")
```

Comment:
This comparison is using nested fields of a struct - consider what would happen if one of those nested fields is `nil`.
For example: if `res.Structure.Keyfamilies` returns `nil` then 
```
res.Structure.Keyfamilies.Keyfamily[index0].Annotations.Annotation[indx].Text.(string)
```
Will panic with a nil pointer error. As this is part of the Dataset API I would suggest we should be careful and handle these possible scenarios carefully.